src/eureka/lib/util.py

```diff
@@ -115,15 +115,18 @@ def check_nans(data, mask, log, name=''):
     """
     data = np.ma.masked_where(mask == 0, np.copy(data))
     num_nans = np.sum(np.ma.masked_invalid(data).mask)
+    num_pixels = np.size(data)
+    perc_nans = 100*num_nans/num_pixels
     if num_nans > 0:
-        log.writelog(f"  WARNING: {name} has {num_nans} NaNs/infs. Your "
-                     "subregion may be off the edge of the detector "
-                     "subarray.\n    Masking NaN region and continuing, "
-                     "but you should really stop and reconsider your "
-                     "choices.")
+        log.writelog(f"  {name} has {num_nans} NaNs/infs, which is "
+                     f"{perc_nans:.2f}% of all pixels.")
         inan = np.where(np.ma.masked_invalid(data).mask)
-        # subdata[inan]  = 0
         mask[inan] = 0
+    if perc_nans > 10:
```

Comment:
Thanks for making this change - the old code was a bit alarmist

Reply:
Agreed!  A little context can go a long way.